sympy/polys/matrices/tests/test_lll.py

```diff
@@ -0,0 +1,142 @@
+from sympy.core.numbers import Rational
+from sympy.polys.domains import ZZ, QQ
+from sympy.polys.matrices import DM
+from sympy.polys.matrices.ddm import DDM
+from sympy.polys.matrices.domainmatrix import DomainMatrix
+from sympy.polys.matrices.exceptions import DMRankError, DMValueError, DMShapeError, DMDomainError
+from sympy.polys.matrices.lll import ddm_lll
+from sympy.testing.pytest import raises
+
+
+def test_lll():
+    def get_normal_test_data():
+        return [
+            (
+                DM(
+                    [
+                        [1, 0, 0, 0, -20160],
+                        [0, 1, 0, 0, 33768],
+                        [0, 0, 1, 0, 39578],
+                        [0, 0, 0, 1, 47757]
+                    ],
+                    ZZ
+                ).rep,
+                DM(
+                    [
+                        [10, -3, -2, 8, -4],
+                        [3, -9, 8, 1, -11],
+                        [-3, 13, -9, -3, -9],
+                        [-12, -7, -11, 9, -1]
+                    ],
+                    ZZ
+                ).rep
+            ),
+            (
+                DM(
+                    [
+                        [20, 52, 3456],
+                        [14, 31, -1],
+                        [34, -442, 0]
+                    ],
+                    ZZ
+                ).rep,
+                DM(
+                    [
+                        [14, 31, -1],
+                        [188, -101, -11],
+                        [236, 13, 3443]
+                    ],
+                    ZZ
+                ).rep
+            ),
+            (
+                DM(
+                    [
+                        [34, -1, -86, 12],
+                        [-54, 34, 55, 678],
+                        [23, 3498, 234, 6783],
+                        [87, 49, 665, 11]
+                    ],
+                    ZZ
+                ).rep,
+                DM(
+                    [
+                        [34, -1, -86, 12],
+                        [291, 43, 149, 83],
+                        [-54, 34, 55, 678],
+                        [-189, 3077, -184, -223]
+                    ],
+                    ZZ
+                ).rep
+            )
+
+        ]
+
+    for basis, reduced in get_normal_test_data():
+        assert ddm_lll(basis, delta=QQ(5, 6)) == reduced
+        assert basis.lll(delta=QQ(5, 6)) == reduced
+        assert basis.to_sdm().lll(delta=QQ(5, 6)) == reduced.to_sdm()
+        assert DomainMatrix.from_rep(basis).lll(delta=Rational(5, 6)).rep == reduced
+
+
+def test_lll_linear_dependent():
+    def get_linear_dependent_test_data():
+        return [
+            DM(
+                [
+                    [0, -1, -2, -3],
+                    [1, 0, -1, -2],
+                    [2, 1, 0, -1],
+                    [3, 2, 1, 0]
+                ],
+                ZZ
+            ).rep,
```

Comment:
There isn't any reason to create a function inside a function to do this. The point was just that we don't want any code to run when the module is imported. If the matrices are created inside this function rather than at top-level in the module then that's fine.

Also This style of code is difficult to parse. I guess it comes from black or something like that (note that SymPy has broadly agreed not to use black).

I would just write this as something like:
```python
mat1 = DM([[0, -1, -2, -3],
           [1, 0, -1, -2],
           [2, 1, 0, -1],
           [3, 2, 1, 0]], ZZ).rep
```
Or perhaps
```python
mat1 = DM([[0, -1, -2, -3],
           [1,  0, -1, -2],
           [2,  1,  0, -1],
           [3,  2,  1,  0]], ZZ).rep
```
Although I'm not sure why we need the `rep`.

I know some tools like `black` want to turn everything into extra indentation and extra lines but to me if you are using a tool like that then basically every time it reformats things it means that you should rewrite the code until it does not need reformatting any more because whenever black decides to change the code it makes it horrible.